### ISSUE TITLE
Automating build testing (online and offline) on various platforms

### ISSUE THREAD

**Issue by Seanny123** [nengo/nengo#478]
I spent a bit of the afternoon checking out how hard it would be to automate a build-testing process on the three major platforms (Linux, Windows, OS X). This point in time, it's possible with Vagrant (VirtualBox automation tool), but not pleasant. Even the testing Gods, the [Travis-CI](https://github.com/travis-ci/travis-ci/issues/2191) folk, are currently figuring out how to do it.

I did find [this tutorial](http://codehenge.net/blog/2013/02/automate-your-development-environment-with-vagrant/) online on how to get started, but that's only a Linux machine. I found [this one](http://www.hurryupandwait.io/blog/peering-into-the-future-of-windows-automation-testing-with-chef-vagrant-and-test-kitchen-look-mom-no-ssh) for a Windows environment, but after (speed) reading through it, there seems like a lot of test-specific things I would have to learn how to use.

In conclusion, maybe hire a co-op student or convince Travis-CI to hire a co-op student? Alternatively, if anyone thinks I just got Google-blinded and found bad links or made bad estimates of time commitment, let me know.

I've posted this problem on [StackOverflow](https://stackoverflow.com/questions/25961576/desktop-cross-platform-build-testing-in-python), but I think the chances of someone asking it is pretty low.

**Comment by Seanny123** [comment]
We're now testing Windows builds on AppVeyor right? Does that mean only OS X is left to automate?

**Comment by jgosmann** [comment]
Not sure if it's worth to automate OS X testing, as it is a Unix flavor the Linux tests should be equivalent in most cases.

**Comment by tbekolay** [comment]
> Not sure if it's worth to automate OS X testing, as it is a Unix flavor the Linux tests should be equivalent in most cases.

Yeah, I'm inclined to agree. If we ever get a Mac OS X specific report that isn't an install issue then we can revisit. Note that TravisCI does have a Mac OS X image, so it's possible to do it, but I think not worth the effort at the moment.

**Comment by Seanny123** [comment]
Cool. I'll close this issue then.

**Comment by jgosmann** [comment]
> Note that TravisCI does have a Mac OS X image

Cool, I did not know this.
